lib/nexpose/site.rb

```diff
@@ -614,9 +614,36 @@ class IPRange
     # End of range *Optional (If nil then IPRange is a single IP Address)
     attr_accessor :to
 
+    # @overload initialize(ip)
+    #   @param [#to_s] from the IP single IP address.
+    #   @example
+    #     Nexpose::IPRange.new('192.168.1.0')
+    #
+    # @overload initialize(start_ip, end_ip)
+    #   @param [#to_s] from the IP to start the range with.
+    #   @param [#to_s] to the IP to end the range with.
+    #   @example
+    #     Nexpose::IPRange.new('192.168.1.0', '192.168.1.255')
+    #
+    # @overload initialize(cidr_range)
+    #   @param [#to_s] from the CIDR notation IP address range.
+    #   @example
+    #     Nexpose::IPRange.new('192.168.1.0/24')
+    #   @note The range will not be stripped of reserved IP addresses (such as
```

Comment:
@mdaines-r7:
>And as to documentation, it should include a mention that this is different from how Nexpose interprets CIDR notation when entered in the UI. In the UI, a CIDR notation gets stripped of its first and last IP in the range. (The UI used to have a message about it, but I don't see it now after the site refactor. But when I test it, it still behaves that way.)
The other alternative is to make it behave like the UI does. Users could still work around it by providing an explicit range.

@jhart-r7:
> I was concerned about this. I agree that the Gem and Nexpose UI should handle the same inputs the same way, but ideally they would support CIDR more naturally (which, IMO, means not stripping the first/last).

@mdaines-r7 @jhart-r7 What do you think about this approach? The `@note` tag renders an obvious block to the user when the HTML documentation is generated.

I personally think we *should* be sending the x.x.x.0 and x.x.x.255 addresses. If Nexpose wants to skip them that's fine but they are valid, assignable, IPs (albeit usually network identifier/broadcast).